scripts/test-network-stalls.go

```diff
@@ -0,0 +1,248 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/zeebo/errs"
+)
+
+var (
+	numTries      = flag.Int("num-tries", 20, "number of tries to cause a hang")
+	bucketName    = flag.String("bucket", "bukkit", "name of bucket to use for test")
+	fileSize      = flag.Int64("file-size", 5*1024*1024, "size of test file to use")
+	deleteTimeout = flag.Duration("timeout", 60*time.Second, "how long to wait for a delete to succeed or time out")
+
+	tryAgain = errs.New("test needs to run again")
+)
+
+func makeRandomContentsFile(path string, size int64) (err error) {
+	inFile, err := os.Open("/dev/urandom")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = errs.Combine(err, inFile.Close())
+	}()
+	outFile, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = errs.Combine(err, outFile.Close())
+	}()
+	if _, err := io.CopyN(outFile, inFile, size); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ErrorWithOutput returns an exit error with all collected output.
+type ErrorWithOutput struct {
+	*exec.ExitError
+	Output []byte
+}
+
+func (e *ErrorWithOutput) Error() string {
+	return e.ExitError.Error() + "\nOutput: " + string(e.Output)
+}
+
+type uplinkRunner struct {
+	execName  string
+	configDir string
+	logLevel  string
+}
+
+func (ur *uplinkRunner) run(ctx context.Context, stdout, stderr io.Writer, args ...string) error {
+	cmdArgs := []string{"--config-dir", ur.configDir, "--log.level", ur.logLevel}
+	cmdArgs = append(cmdArgs, args...)
+	cmd := exec.CommandContext(ctx, ur.execName, cmdArgs...)
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+	return cmd.Run()
+}
+
+func (ur *uplinkRunner) GetOutputCtx(ctx context.Context, args ...string) ([]byte, error) {
+	var buf bytes.Buffer
+	err := ur.run(ctx, &buf, &buf, args...)
+	return buf.Bytes(), err
+}
+
+func (ur *uplinkRunner) GetOutput(args ...string) ([]byte, error) {
+	return ur.GetOutputCtx(context.Background(), args...)
+}
+
+func (ur *uplinkRunner) CallCtx(ctx context.Context, args ...string) error {
+	output, err := ur.GetOutputCtx(ctx, args...)
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			return &ErrorWithOutput{exitErr, output}
+		}
+		return err
+	}
+	return nil
+}
+
+func (ur *uplinkRunner) Call(args ...string) error {
+	return ur.CallCtx(context.Background(), args...)
+}
+
+// skip the first four whitespace-delimited fields and keep the rest
+var lsOutputRegexp = regexp.MustCompile(`(?m)^\s*(?:\S+\s+){4}(.*)$`)
+
+func (ur *uplinkRunner) doesRemoteExist(remotePath string) (bool, error) {
+	pathParts := strings.Split(remotePath, "/")
```

Comment:
`path.Split`

Reply:
that would work fine here, but I'm hesitant to use a URL-type string with path.Split, just because of potential confusion (you couldn't correctly use path.Join to reassemble the parts).